android/engine/src/test/java/org/smartregister/fhircore/engine/task/FhirCarePlanGeneratorTest.kt

```diff
@@ -723,6 +739,195 @@ class FhirCarePlanGeneratorTest : RobolectricTest() {
       }
   }
 
+  @Test
+  fun testGenerateCarePlanForANCVisitWithCql() = runTest {
+    val plandefinition =
+      "plans/anc-visit/plandefinition-apply.json".readFile().decodeResourceFromString<PlanDefinition>()
+
+    val patient =
+      "plans/anc-visit/sample/patient.json".readFile().decodeResourceFromString<Patient>()
+
+    val questionnaireResponse =
+      "plans/anc-visit/sample/questionnaire-response-register.json"
+        .readFile()
+        .decodeResourceFromString<QuestionnaireResponse>()
+
+    val cql =
+      "plans/anc-visit/library.cql"
+        .readFile()
+
+    val translator = compileCql(cql)
+
+    val attachmentCql =
+      cql.let {
+        Attachment().apply {
+          contentType = "text/cql"
+          data = it.toByteArray()
+        }
+      }
+
+    val attachmentJson =
+      translator.toJson()?.let {
+        Attachment().apply {
+          contentType = "application/elm+json"
+          data = it.toByteArray()
+        }
+      }
+
+    val attachmentXml =
+      translator.toXml()?.let {
+        Attachment().apply {
+          contentType = "application/elm+xml"
+          data = it.toByteArray()
+        }
+      }
+
+    val library = translator.translatedLibrary.let { lib ->
+      Library().apply {
+        id = lib.identifier.id
+        name = lib.identifier.id
+        version = lib.identifier.version
+        status = Enumerations.PublicationStatus.ACTIVE
+        experimental = true
+        url = "/Library/${lib.identifier.id}|${lib.identifier.version}"
+        attachmentCql?.let { addContent(it) }
+        attachmentJson?.let { addContent(it) }
+        attachmentXml?.let { addContent(it) }
+      }
+    }
+
+    fhirOperator.loadLib(library)
+
+    // start of plan is lmp date | set lmp date to 4 months , and 15th of month
+    val lmp = Date().plusMonths(-4).apply { date = 15 }
```

Comment:
do we need to test other variations on the lmp date to hit other code-paths / edge cases?